src/frequenz/client/electricity_trading/_client.py

```diff
@@ -799,58 +799,57 @@ async def list_gridpool_orders(
             delivery_area: The delivery area to filter by.
             tag: The tag to filter by.
-            max_nr_orders: The maximum number of orders to return.
-            page_token: The page token to use for pagination.
+            page_size: The number of orders to return per page.
             timeout: Timeout duration, defaults to None.
 
-        Returns:
-            The list of orders for that gridpool.
+        Yields:
+            The list of orders for the given gridpool.
 
         Raises:
             grpc.RpcError: If an error occurs while listing the orders.
         """
-        gridpool_order_filer = GridpoolOrderFilter(
+        gridpool_order_filter = GridpoolOrderFilter(
             order_states=order_states,
             side=side,
             delivery_period=delivery_period,
             delivery_area=delivery_area,
             tag=tag,
         )
 
-        pagination_params = Params(
-            page_size=max_nr_orders,
-            page_token=page_token,
+        request = electricity_trading_pb2.ListGridpoolOrdersRequest(
+            gridpool_id=gridpool_id,
+            filter=gridpool_order_filter.to_pb(),
+            pagination_params=(
+                Params(page_size=page_size).to_proto() if page_size else None
+            ),
         )
-
-        try:
-            response = await cast(
-                Awaitable[electricity_trading_pb2.ListGridpoolOrdersResponse],
-                grpc_call_with_timeout(
-                    self.stub.ListGridpoolOrders,
-                    electricity_trading_pb2.ListGridpoolOrdersRequest(
-                        gridpool_id=gridpool_id,
-                        filter=gridpool_order_filer.to_pb(),
-                        pagination_params=pagination_params.to_proto(),
+        while True:
+            try:
+                response = await cast(
+                    Awaitable[electricity_trading_pb2.ListGridpoolOrdersResponse],
+                    grpc_call_with_timeout(
+                        self.stub.ListGridpoolOrders,
+                        request,
+                        metadata=self._metadata,
+                        timeout=timeout,
                     ),
-                    metadata=self._metadata,
-                    timeout=timeout,
-                ),
-            )
+                )
 
-            orders: list[OrderDetail] = []
-            for order_detail in response.order_details:
-                try:
-                    orders.append(OrderDetail.from_pb(order_detail))
-                except InvalidOperation:
-                    _logger.error(
-                        "Failed to convert order details for order: %s",
-                        str(order_detail).replace("\n", ""),
+                for order_detail in response.order_details:
+                    yield OrderDetail.from_pb(order_detail)
+
+                if len(response.pagination_info.next_page_token):
```

Comment:
You don't need the `len` here. An empty string is false.

Reply:
done